src/main/java/org/broadinstitute/hellbender/tools/walkers/qc/Pileup.java

```diff
@@ -0,0 +1,195 @@
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, doc = "An output file created by the walker. Will overwrite contents if file exists")
+    public PrintStream out;
+
+    /**
+     * In addition to the standard pileup output, adds 'verbose' output too. The verbose output contains the number of
+     * spanning deletions, and for each read in the pileup it has the read name, offset in the base string, read length,
+     * and read mapping quality.  These per read items are delimited with an '@' character.
+     */
+    @Argument(fullName = "showVerbose", shortName = "verbose", doc = "Add an extra verbose section to the pileup output", optional = true)
+    public boolean showVerbose = false;
+
+    /**
+     * This enables annotating the pileup to show overlaps with metadata from a Feature file(s). For example, if you provide a
+     * VCF and there is a SNP at a given location covered by the pileup, the pileup output at that position will be
+     * annotated with the corresponding source Feature identifier.
+     */
+    @Argument(fullName = "metadata", shortName = "metadata", doc = "Features file(s) containing metadata", optional = true)
+    public List<FeatureInput<Feature>> metadata = new ArrayList<>();
+
+    /**
+     * Adds the length of the insert each base comes from to the output pileup. Here, "insert" refers to the DNA insert
+     * produced during library generation before sequencing.
+     */
+    @Hidden
+    @Argument(fullName = "outputInsertLength", shortName = "outputInsertLength", doc = "Output insert length", optional = true)
+    public boolean outputInsertLength = false;
+
+    @Override
+    public boolean includeDeletions() {
+        return false;
+    }
+
+    @Override
+    public CountingReadFilter makeReadFilter(){
+        return new CountingReadFilter("Wellformed", new WellformedReadFilter(getHeaderForReads()))
+            .and(new CountingReadFilter("Mapped", ReadFilterLibrary.MAPPED))
+            .and(new CountingReadFilter("Not duplicate", ReadFilterLibrary.NOT_DUPLICATE))
+            .and(new CountingReadFilter("Passes VQ check", ReadFilterLibrary.PASSES_VENDOR_QUALITY_CHECK))
+            .and(new CountingReadFilter("Primary alignment", ReadFilterLibrary.PRIMARY_ALIGNMENT));
+    }
+
+    @Override
+    public void apply(AlignmentContext alignmentContext, ReferenceContext referenceContext, FeatureContext featureContext) {
+        final String features = getFeaturesString(featureContext);
+        final ReadPileup basePileup = alignmentContext.getBasePileup();
+        final StringBuilder s = new StringBuilder();
+        s.append(String.format("%s %s",
+                basePileup.getPileupString((hasReference()) ? (char) referenceContext.getBase() : 'N'),
+                features));
+        if (outputInsertLength) {
+            s.append(" ").append(insertLengthOutput(basePileup));
```

Comment:
add a test case that reaches this line ` s.append(" ").append(insertLengthOutput(basePileup));`